Objects/setobject.c

```diff
@@ -2424,50 +2564,75 @@ PySet_Size(PyObject *anyset)
         PyErr_BadInternalCall();
         return -1;
     }
-    return PySet_GET_SIZE(anyset);
+    return set_len((PySetObject *)anyset);
 }
 
 int
 PySet_Clear(PyObject *set)
 {
+    int rv;
+
     if (!PySet_Check(set)) {
         PyErr_BadInternalCall();
         return -1;
     }
-    return set_clear_internal((PySetObject *)set);
+
+    Py_BEGIN_CRITICAL_SECTION(set);
+    rv = set_clear_internal((PySetObject *)set);
+    Py_END_CRITICAL_SECTION();
+    return rv;
```

Comment:
This is fine, but you can also call `set_clear((PySetObject *)set, NULL);`, which does the locking internally and so doesn't need extra `Py_BEGIN_CRITICAL_SECTION`.